WalletWasabi/Models/Label.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace WalletWasabi.Models
+{
+	public class Label : IEquatable<Label>
+	{
+		public static Label Empty { get; } = new Label();
+		public static char[] Separators { get; } = new[] { ',', ':' };
+		public IEnumerable<string> Labels { get; }
+		public bool IsEmpty { get; }
+
+		private string LabelString { get; }
+
+		public Label(params string[] labels) : this(labels as IEnumerable<string>)
+		{
+		}
+
+		public Label(IEnumerable<string> labels)
+		{
+			Labels = (labels
+				   ?.SelectMany(x => x?.Split(Separators, StringSplitOptions.RemoveEmptyEntries) ?? new string[0])
+				   ?.Select(x => x.Trim())
+				   ?.Where(x => x != "")
+				   ?.Distinct(StringComparer.OrdinalIgnoreCase)
+				   ?.OrderBy(x => x)
+				   ?? Enumerable.Empty<string>())
+				   .ToArray();
+
+			HashCode = ((IStructuralEquatable)Labels).GetHashCode(EqualityComparer<string>.Default);
+			IsEmpty = !Labels.Any();
+
+			LabelString = IsEmpty ? "" : string.Join(", ", Labels);
```

Comment:
I think this is not really necessary.
```suggestion
			LabelString = string.Join(", ", Labels);
```

Reply:
Correct.